service/src/ServiceIOGroup.cpp

```diff
@@ -384,10 +388,14 @@ namespace geopm
                                                   m_control_requests,
                                                   server_pid,
                                                   server_key);
-            m_batch_server = BatchClient::make_unique(server_key,
-                                                      1.0,
-                                                      m_signal_requests.size(),
-                                                      m_control_requests.size());
+            if (m_batch_server == nullptr) {
```

Comment:
change m_batch_server variable name to m_batch_client (it is confusing since this is the client interface)